File: ai/asriel/test-xor.py
import qortex as Q
import numpy as np

xor = Q.Supervised(2, 4, Q.Cell([
	Q.Transform(4),
	Q.Transform(1, "linear")
]))

print dir(xor.runop)
exit()

jar = Q.Jar()

ex = np.array([[0, 0, 1, 1], [0, 1, 0, 1]])
ts = np.array([0, 1, 1, 0])

epoch = 0
while True:
	print xor.train(ex, ts, jar)
	epoch += 1

File: ai/asriel/qortex/test-xor.cpp
#include <iostream>

#include "qortex.hpp"

using namespace Q;

int main() {
	Cell cell;
	cell.add(
		new Transform(2, "middle"),
		new Transform(1, "end")
	);
	cell.build(2);
	
	Batch in(2, 4);
	in <<
		-1, -1, 1, 1,
		-1, 1, -1, 1;
	
	Batch t(1, 4);
	t << -1, 1, 1, -1;
	
	int epoch = 0;
	while(true) {
		auto p = cell.prepare(in);
		cell.train(t - p);
		
		if(++epoch%1024 == 0) {
			std::cout << cell.run(in) << std::endl;
		}
	}
}
